packages/rx_bloc_cli/mason_templates/bricks/feature_onboarding/__brick__/lib/feature_onboarding_phone_confirmation/repositories/search_country_code_repository.dart

```diff
@@ -0,0 +1,22 @@
+{{> licence.dart }}
+
+import 'dart:async';
+
+import '../../base/data_sources/remote/country_codes_remote_data_source.dart';
+import '../../base/models/country_code_model.dart';
+
+class SearchCountryCodeRepository<T> {
+  SearchCountryCodeRepository(this._countryCodesRemoteDataSource) {
+    _getCountryCodes();
+  }
+
+  final CountryCodesRemoteDataSource _countryCodesRemoteDataSource;
+  final _completer = Completer<List<CountryCodeModel>>();
+
+  Future<void> _getCountryCodes() async {
+    final countryCodes = await _countryCodesRemoteDataSource.getCountryCodes();
+    _completer.complete(countryCodes['countryCodes']);
+  }
+
+  Future<List<T>> get searchList => _completer.future as Future<List<T>>;
```

Comment:
What will happen if the `future` is not of the `Future<List<T>>` type ?

Reply:
I still do not understand why we need generic here. Could you please elaborate ?

Reply:
This was copied from a similar repository then adjusted. Not sure why the types have been set to generic ones in the repository, but now I've updated them.